### ISSUE TITLE
Warn on usage of HTTP links in SETUP

### ISSUE THREAD

**Issue by madhuracj** [phpmyadmin/phpmyadmin#12468]
As reported by Emanuel Bronshtein,
Warn on usage of HTTP links (not in case of onion domains / localhost / localhost IPs ) in SETUP in the following directives:
    $cfg['Servers'][$i]['LogoutURL']
    $cfg['Servers'][$i]['SignonURL']
    $cfg['NavigationLogoLink']

**Comment by williamdes** [comment]
@emanuelb Could you detail how this issue stil affects us and how you see it fixed ?

**Comment by emanuelb** [comment]
Yes, visiting:
https://demo.phpmyadmin.net/QA_5_0/setup/index.php?page=form&formset=Sql
and entering "http://example.com/logo.png" for example in "Logo link URL", clicking "Apply"
should generate a warring (which is happening for other configurations) as insecure HTTP uri is used, same for any other configuration that can accept HTTP uri. 
Insecure HTTP uri is URL that will be vulnerable to MITM attack, which exclude:
.onion domains (as TOR protect against MITM)
localhost and localhost IPs (doesn't involve the netwotk)
otherwise it's not safe.

**Comment by williamdes** [comment]
Thank you !
Do you have a list of the RFC local domains ?

**Comment by emanuelb** [comment]
no, but there is a regex in answer here for example that cover loopback addresses:
https://stackoverflow.com/questions/8426171/what-regex-will-match-all-loopback-addresses
The php filter_var function doesn't have this option, only FILTER_FLAG_NO_RES_RANGE which also cover stuff that should-not be matched (such as 169.254.0.0/16 range).

code example, not tested, to give idea how the check can look like.
```php
if(parse_url($url, PHP_URL_SCHEME) === 'http') {
$hostname_in_url = parse_url($url, PHP_URL_HOST);
if(!
// regex from above
|| strtolower(substr($hostname_in_url, -6)) === '.onion' 
) { 
// generate warning here
}}
```

There are more places that can implement this check, for example:
visit:
https://demo.phpmyadmin.net/QA_5_0/setup/index.php
click "New server", click "Apply".
result in warning:
> Use SSL (localhost [1])
You should use SSL connections if your database server supports it. 

The warning for not using SSL is good, just not correct if the hostname is loopback address (here just use the Regex, as it's not url but a hostname/ip value field)

The original report is for adding such warning (call it "Use HTTPS", "You should use HTTPS instead of HTTP...") when insecure http:// uri is used in values for:
```
$cfg['Servers'][$i]['LogoutURL']
$cfg['Servers'][$i]['SignonURL']
$cfg['NavigationLogoLink']
```
